.github/workflows/sage-test.yml

```diff
@@ -0,0 +1,149 @@
+name: Sage Install Tests
+on:
+  push:
+    paths:
+      - '.github/workflows/sage-test.yml'
+  pull_request:
+    branches:
+      - main
+    paths:
+      - 'private/scripts/**'
+jobs:
+  test:
+    name: Sage Install Tests
+    env:
+      TERM: xterm-256color
+    runs-on: ${{ matrix.os }}
+    defaults:
+      run:
+        shell: bash
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+        php-version: [8.0, 8.1, 8.2, 8.3]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Install SSH keys
+        uses: webfactory/ssh-agent@v0.8.0
+        with:
+          ssh-private-key: ${{ secrets.SSH_PRIVATE_KEY }}
+      - name: Get and Set PR number
+        id: pr
+        run: |
+          pr_num=$(gh pr view --json number -q .number || echo "")
+          echo "pr_num=$pr_num" >> $GITHUB_ENV
+        env:
+          GITHUB_TOKEN: ${{ github.token }}
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php-version }}
+      - name: Generate multidev name
+        id: generate_name
+        run: |
+          os_short=""
+          case "${{ matrix.os }}" in
+            ubuntu-latest)
+              os_short="nix"
+              ;;
+            macos-latest)
+              os_short="mac"
+              ;;
+            windows-latest)
+              os_short="win"
+              ;;
+          esac
+
+          php_short=$(echo "${{ matrix.php-version }}" | tr -d '.')
+          echo "PR number: $pr_num"
+          echo "OS: $os_short"
+          echo "PHP: $php_short"
+          multidev_name="${os_short}-${php_short}-${pr_num}"
+          echo "Generated multidev name: $multidev_name"
+          echo "multidev_name=$multidev_name" >> $GITHUB_ENV
+      - name: Get latest Terminus release
+        run: |
+          latest_version=$(curl --silent "https://api.github.com/repos/pantheon-systems/terminus/releases/latest" | jq -r .tag_name)
+          echo "Latest Terminus version: $latest_version"
+          if [[ "${{ matrix.os }}" == "macos-latest" ]]; then
+            echo "☕ Installing Terminus on macOS from Homebrew..."
+            brew install pantheon-systems/external/terminus
+          else
+            echo "💻 Installing Terminus from phar on Linux..."
+            curl -L "https://github.com/pantheon-systems/terminus/releases/download/$latest_version/terminus.phar" -o terminus
```

Comment:
github should have a latest release endpoint so you needn't look it up.

Optional.